### PR TITLE
Shields panel webui resource

### PR DESCRIPTION
**Pull request by nullhook**
Resolves https://github.com/brave/brave-browser/issues/18785

~DO NOT MERGE BEFORE https://github.com/brave/brave-core/pull/10522~ ✅ 

Visit: `chrome://brave-shields.top-chrome` 
with flag `--enable-features=BraveShieldsPanelV2`

image
image
image

## Submitter Checklist:

- [ ] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: browser/ui/views/brave_actions/brave_shields_action_view.cc [hunk 1]
```diff
@@ -9,7 +9,10 @@
 #include <string>
 #include <utility>
 
+#include "base/logging.h"
```
Comment (nullhook) [inline browser/ui/views/brave_actions/brave_shields_action_view.cc:12]: @simonhong can be removed?

File: components/brave_shields/resources/panel/components/main-panel/index.tsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
+import * as React from 'react'
+
+import * as S from './style'
+import Toggle from '../../../../../web-components/toggle'
+import { CaratStrongDownIcon } from 'brave-ui/components/icons'
+import AdvancedControlsContent from '../advanced-controls-content'
+import { getLocale } from '../../../../../common/locale'
+
+function MainPanel () {
+  const [isAdvCtrlsExpanded, setIsisAdvCtrlsExpanded] = React.useState(true)
```
Comment (nullhook) [inline components/brave_shields/resources/panel/components/main-panel/index.tsx:10]: this is set to true by default for development only

File: components/web-components/toggle/style.ts [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+  ${p => p.size && matchForSize(p.size)}
+  ${p => p.isActive && matchForActiveBrandColor(p.brand)}
+
+  &:hover {
+    --border-color: ${p => p.theme.color.interactive05};
+  }
+
+  &:focus {
+    --border-color: ${p => p.theme.color.focusBorder};
+  }
```
Comment (petemill) [inline components/web-components/toggle/style.ts:131]: I think we're missing the `:active` state aka "pressed" state in figma - which seems to be the same as default, so undoes the hover / focus style.

File: components/web-components/toggle/index.tsx [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+  }
+
+  return (
+    <S.ToggleBox
+      type='button'
+      role='switch'
+      aria-checked={isOn}
+      aria-label={props.accessibleLabel}
+      onClick={onToggleClick}
+      isActive={isOn}
```
Comment (petemill) [inline components/web-components/toggle/index.tsx:32]: I'm wondering what the reason to have 2 concepts of 'checked' - `isOn` and `isActive`? Should we stick with one of them? To me, 'active' might conflict with the ':active' css pseudo class, and I don't think it's being used to represent that here.

File: components/web-components/toggle/style.ts [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+}
+
+export const ToggleBox = styled.button<Props>`
+  --bg-color: ${(p) => p.theme.color.disabled};
+  --border-color: transparent;
+  --border-width: 2px;
+  --animation-name: none;
+
+  --width: 60px;
+  --height: 30px;
```
Comment (petemill) [inline components/web-components/toggle/style.ts:90]: nit: I think maybe too many css variables here, which is making it more difficult to scan the code since there's a separation of definitions and not 100% consistency (e.g. border-radius does not use a css var). Perhaps we should restrict to properties which are either:
- used more than once, in other properties or calculations
- modified by some state change (e.g. psuedo class)

**Review by petemill — CHANGES_REQUESTED** [review]
Just reviewed the knob so far

File: browser/brave_content_browser_client.cc [hunk 6]
```diff
@@ -190,6 +190,11 @@ using extensions::ChromeContentBrowserClientExtensionsPart;
 #include "brave/browser/brave_ads/brave_ads_host.h"
 #endif  // defined(OS_ANDROID)
 
+#if !defined(OS_ANDROID)
+#include "brave/browser/ui/webui/brave_shields/shields_panel_ui.h"
```
Comment (simonhong) [inline browser/brave_content_browser_client.cc:194]: Let's use above `#if !defined(OS_ANDROID)` section instead of adding same define.

File: browser/ui/BUILD.gn [hunk 7]
```diff
@@ -260,6 +264,8 @@ source_set("ui") {
     "//brave/components/brave_rewards/resources:tip_generated_resources",
     "//brave/components/brave_search/common",
     "//brave/components/brave_shields/browser",
+    "//brave/components/brave_shields/common:mojom",
```
Comment (simonhong) [inline browser/ui/BUILD.gn:267]: I think all new deps should be added in below `if (!is_android) `?

File: browser/ui/webui/brave_shields/shields_panel_handler.h [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+#include "mojo/public/cpp/bindings/receiver.h"
+#include "ui/webui/mojo_bubble_web_ui_controller.h"
+
+namespace content {
+class WebUI;
+}  // namespace content
+
+class ShieldsPanelHandler : public brave_shields_panel::mojom::PanelHandler {
+ public:
+  using GetWebContentsForTabCallback =
```
Comment (simonhong) [inline browser/ui/webui/brave_shields/shields_panel_handler.h:23]: Who use this?

File: browser/ui/webui/brave_shields/shields_panel_ui.cc [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+
+  webui::SetupWebUIDataSource(source,
+                              base::make_span(kBraveShieldsPanelGenerated,
+                                              kBraveShieldsPanelGeneratedSize),
+                              IDR_SHIELDS_PANEL_HTML);
+
+  content::WebUIDataSource::Add(web_ui->GetWebContents()->GetBrowserContext(),
+                                source);
+  Profile* profile = Profile::FromWebUI(web_ui);
+  content::URLDataSource::Add(
```
Comment (simonhong) [inline browser/ui/webui/brave_shields/shields_panel_ui.cc:37]: I think we don't need `FaviconSource` for shields webui?

File: components/brave_shields/common/BUILD.gn [hunk 10]
```diff
@@ -14,12 +14,20 @@ static_library("common") {
...
-  sources = [ "brave_shields.mojom" ]
+  sources = [
+    "brave_shields.mojom",
+    "brave_shields_panel.mojom",
+  ]
 
-  deps = [ "//mojo/public/mojom/base" ]
+  public_deps = [
+    "//mojo/public/mojom/base",
+    "//url/mojom:url_mojom_gurl",
```
Comment (simonhong) [inline components/brave_shields/common/BUILD.gn:31]: Using `public_deps` here seems good but why did you add `url_mojom_gurl`?
It seems shields panel doesn't use this

File: components/brave_shields/resources/panel/components/advanced-controls-content/index.tsx [hunk 11]
```diff
@@ -0,0 +1,140 @@
...
+import * as S from './style'
+import { CaratStrongDownIcon } from 'brave-ui/components/icons'
+import Toggle from '../../../../../web-components/toggle'
+import { getLocale } from '../../../../../common/locale'
+
+function GlobalSettings () {
+  return (
+    <S.SettingsBox>
+      <S.SettingsTitle>{getLocale('braveShieldsGlobalSettingsTitle')}</S.SettingsTitle>
+      <a href='brave://settings/shields' target='_blank'>
```
Comment (simonhong) [inline components/brave_shields/resources/panel/components/advanced-controls-content/index.tsx:12]: This doesn't work. In webui, we should use `chrome` scheme.
and anchor tag doesn't work in panel.

File: components/brave_shields/resources/panel/components/advanced-controls-content/index.tsx [hunk 12]
```diff
@@ -0,0 +1,140 @@
...
+        <S.GlobeIcon />
+        {getLocale('braveShieldsChangeDefaults')}
+      </a>
+      <a href='brave://adblock' target='_blank'>
```
Comment (simonhong) [inline components/brave_shields/resources/panel/components/advanced-controls-content/index.tsx:16]: ditto

File: components/brave_shields/common/brave_shield_constants.h [hunk 13]
```diff
@@ -38,6 +40,46 @@ const char kObsoleteFingerprinting[] = "fingerprinting";
 const base::FilePath::CharType kCustomSubscriptionListText[] =
     FPL("list_text.txt");
 
+constexpr webui::LocalizedString kLocalizedStrings[] = {
```
Comment (petemill) [inline components/brave_shields/common/brave_shield_constants.h:43]: Is this file used by both android and desktop? I haven't looked, but If so, we should wrap this so that it isn't compiled on android as it already has its own strings.

File: components/brave_shields/resources/panel/components/advanced-controls-content/style.ts [hunk 14]
```diff
@@ -0,0 +1,102 @@
...
+
+export const SettingsDesc = styled(SettingsTitle)`
+  color: ${(p) => p.theme.color.text03};
+`
+
+export const GlobeIcon = styled.i`
+  display: block;
+  width: 17px;
+  height: 17px;
+  background-image: url(${globalIconUrl});
```
Comment (petemill) [inline components/brave_shields/resources/panel/components/advanced-controls-content/style.ts:38]: how about inline svg? Looks like links are mean to change color on hover anyway.

File: components/brave_shields/resources/panel/components/main-panel/style.ts [hunk 15]
```diff
@@ -0,0 +1,134 @@
...
+  font-size: 14px;
+  color: var(--text-color);
+  width: 100%;
+  padding: 14px var(--pad-x);
+  border: var(--border);
+  display: flex;
+  align-items: center;
+  justify-content: space-between;
+
+  svg {
```
Comment (petemill) [inline components/brave_shields/resources/panel/components/main-panel/style.ts:69]: This is assuming that the react components coming from the icon library are `svg`. Perhaps it would be more stable to use the common pattern of wrapping with an element to restrict the width / height and perform transform. Or if you really don't want multiple components then extending the style via `Styled(CaratStrongDownIcon)`, if that works.

File: components/brave_shields/resources/panel/theme/shields-dark.ts [hunk 16]
```diff
@@ -0,0 +1,20 @@
...
+ import IThemeShields from './shields-theme'
+ import defaultDarkTheme from 'brave-ui/theme/brave-dark'
+
+ const shieldsDarkTheme: ITheme & IThemeShields = {
+   ...defaultDarkTheme,
+   name: 'Shields Dark',
+   color: {
+     ...defaultDarkTheme.color,
+     background03: '#3B3E4F',
+     text01: '#F0F2FF'
```
Comment (petemill) [inline components/brave_shields/resources/panel/theme/shields-dark.ts:16]: Do these need to differ from the base theme? If the designer has labelled it as text01 then I'm confused why it would be different than the base, unless our base is still incorrect?

File: components/web-components/toggle/index.tsx [hunk 17]
```diff
@@ -0,0 +1,42 @@
+import * as React from 'react'
+
+import * as S from './style'
+
+export type Size = 'sm' | 'lg'
+export type Brand = 'vpn' | 'shields'
```
Comment (petemill) [inline components/web-components/toggle/index.tsx:6]: hmm, in the design system it seems like the "fabulously large toggle" (FLT) is the only version that has a difference between shields and vpn versions, and is quite different from the default and small variants which always have purple highlights, apart from when it's the "colored" version and it's always white highlight. The FLT always has white highlights, like the "colored" version, but has the branded gradient backrounds.

Perhaps we should split the FLT to a different export and add the Brand prop to that? That would more closely resemble the Figma library:

```ts
interface ToggleProps {
  size?: 'sm' | 'default'
  theme?: 'default' | 'colored' // default is automatically the light or dark version
}

interface FabulouslyLargeToggleProps extends ToggleProps {
  brand?: 'vpn' | 'shields'
}
```

**Review by petemill — CHANGES_REQUESTED** [review]
Great start to the new shields UI! I've left some feedback and questions in addition to @simonhong's.

File: components/web-components/toggle/style.ts [hunk 18]
```diff
@@ -0,0 +1,151 @@
...
+  `}
+
+  ${p => p.size && matchForSize(p.size)}
+  ${p => p.isActive && matchForActiveBrandColor(p.brand)}
+
+  &:hover {
+    --border-color: ${p => p.theme.color.interactive05};
+  }
+
+  &:focus {
```
Comment (petemill) [inline components/web-components/toggle/style.ts:129]: I think we need to use `:focus-visible` for everything instead of `:focus` especially since this is a button element so I assume will get focus style applied when using mouse button

File: browser/ui/webui/brave_shields/shields_panel_handler.h (hunk 8 above)
Comment (nullhook) [reply browser/ui/webui/brave_shields/shields_panel_handler.h:23]: not sure actually. copied from vpn.

File: components/brave_shields/resources/panel/theme/shields-dark.ts (hunk 16 above)
Comment (nullhook) [reply components/brave_shields/resources/panel/theme/shields-dark.ts:16]: Yes, our base has incorrect color for darkmode for text01.

Also, on Figma, the background in light mode points to: background02 but in dark mode its: background03

File: components/brave_shields/resources/panel/components/advanced-controls-content/style.ts (hunk 14 above)
Comment (nullhook) [reply components/brave_shields/resources/panel/components/advanced-controls-content/style.ts:38]: Can we do this as a follow up? Would be more elegant after adding the webpack plugin. SVGs tend to be long, inlining can pollute the file.

File: browser/ui/webui/brave_shields/shields_panel_ui.cc (hunk 9 above)
Comment (nullhook) [reply browser/ui/webui/brave_shields/shields_panel_ui.cc:37]: yes, not needed now.

File: components/brave_shields/common/BUILD.gn (hunk 10 above)
Comment (nullhook) [reply components/brave_shields/common/BUILD.gn:31]: my fault, will remove

File: components/web-components/toggle/index.tsx (hunk 17 above)
Comment (nullhook) [reply components/web-components/toggle/index.tsx:6]: makes sense.

File: components/brave_shields/common/brave_shield_constants.h (hunk 13 above)
Comment (nullhook) [reply components/brave_shields/common/brave_shield_constants.h:43]: doesnt seem to be used in android with a quick ripgrep search of the header file. added flag.

File: browser/ui/webui/brave_shields/shields_panel_handler.cc [hunk 19]
```diff
@@ -0,0 +1,39 @@
...
+
+void ShieldsPanelHandler::CloseUI() {
+  auto embedder = webui_controller_->embedder();
+  if (embedder) {
+    embedder->CloseUI();
+  }
+}
+
+void ShieldsPanelHandler::OpenURL(const GURL& url) {
+  Browser* browser = chrome::FindLastActive();
```
Comment (simonhong) [inline browser/ui/webui/brave_shields/shields_panel_handler.cc:35]: This can be improved with browser finder as mentioned in DM.

File: browser/ui/webui/brave_shields/shields_panel_handler.cc [hunk 20]
```diff
@@ -0,0 +1,39 @@
+// Copyright (c) 2021 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include "brave/browser/ui/webui/brave_shields/shields_panel_handler.h"
+
+#include <utility>
+#include "chrome/browser/ui/browser_finder.cc"
```
Comment (simonhong) [inline browser/ui/webui/brave_shields/shields_panel_handler.cc:9]: `browser_finder.h`

File: browser/ui/webui/brave_shields/shields_panel_handler.cc [hunk 21]
```diff
@@ -0,0 +1,39 @@
+// Copyright (c) 2021 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include "brave/browser/ui/webui/brave_shields/shields_panel_handler.h"
+
+#include <utility>
```
Comment (simonhong) [inline browser/ui/webui/brave_shields/shields_panel_handler.cc:8]: needs blank line between std header including and others.

File: browser/ui/webui/brave_shields/shields_panel_handler.h [hunk 22]
```diff
@@ -0,0 +1,41 @@
+// Copyright (c) 2021 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at http://mozilla.org/MPL/2.0/.
+
+#ifndef BRAVE_BROWSER_UI_WEBUI_BRAVE_SHIELDS_SHIELDS_PANEL_HANDLER_H_
+#define BRAVE_BROWSER_UI_WEBUI_BRAVE_SHIELDS_SHIELDS_PANEL_HANDLER_H_
+
+#include <string>
+#include <vector>
```
Comment (simonhong) [inline browser/ui/webui/brave_shields/shields_panel_handler.h:10]: vector and string seems not used?

File: browser/ui/webui/brave_shields/shields_panel_handler.h [hunk 23]
```diff
@@ -0,0 +1,41 @@
...
+
+#include "brave/components/brave_shields/common/brave_shields_panel.mojom.h"
+#include "mojo/public/cpp/bindings/pending_receiver.h"
+#include "mojo/public/cpp/bindings/receiver.h"
+#include "ui/webui/mojo_bubble_web_ui_controller.h"
```
Comment (simonhong) [inline browser/ui/webui/brave_shields/shields_panel_handler.h:15]: This can be replaced with forward decls.

File: browser/ui/webui/brave_shields/shields_panel_ui.cc [hunk 24]
```diff
@@ -0,0 +1,59 @@
+// Copyright (c) 2021 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include "brave/browser/ui/webui/brave_shields/shields_panel_ui.h"
+
+#include <memory>
```
Comment (simonhong) [inline browser/ui/webui/brave_shields/shields_panel_ui.cc:8]: can be deleted as header already includes this.

File: browser/ui/views/brave_actions/brave_shields_action_view.h [hunk 25]
```diff
@@ -26,6 +28,8 @@ class BraveShieldsActionView : public views::LabelButton {
   SkPath GetHighlightPath() const;
 
  private:
+  std::unique_ptr<WebUIBubbleManagerT<ShieldsPanelUI>> webui_bubble_manager_;
+  Profile* profile_ = nullptr;
```
Comment (simonhong) [inline browser/ui/views/brave_actions/brave_shields_action_view.h:32]: Locating member variables after member method is convention.

**Review by simonhong — APPROVED** [review]

File: components/brave_shields/resources/panel/components/advanced-controls-content/style.ts (hunk 14 above)
Comment (petemill) [reply components/brave_shields/resources/panel/components/advanced-controls-content/style.ts:38]: Yes I meant inline via a separate component like we do for brave-ui icons, or otherwise a separate component. Follow-up is ok I suppose, but can also just use [svg2jsx](https://svg2jsx.com/) and then follow-up with a plugin if you like.

File: components/brave_shields/resources/panel/theme/shields-dark.ts (hunk 16 above)
Comment (petemill) [reply components/brave_shields/resources/panel/theme/shields-dark.ts:16]: That's weird - we should let design know when we see inconsistency like that. Aware we can't for this project, so ok on the separate theme override.

File: components/brave_shields/resources/panel/.tsconfig.json [hunk 26]
```diff
@@ -0,0 +1,11 @@
+{
+	"extends": "../../../../tsconfig",
```
Comment (petemill) [inline components/brave_shields/resources/panel/.tsconfig.json:2]: nit: tabs instead of spaces?

File: components/brave_shields/resources/panel/components/advanced-controls-content/index.tsx [hunk 27]
```diff
@@ -0,0 +1,153 @@
...
+    e.preventDefault()
+    const target = e.target as HTMLAnchorElement
+    getPanelBrowserAPI().panelHandler.openURL({ url: target.href })
+  }
+
+  return (
+    <S.SettingsBox>
+      <S.SettingsTitle>{getLocale('braveShieldsGlobalSettingsTitle')}</S.SettingsTitle>
+      <a
+        href='brave://settings/shields'
```
Comment (petemill) [inline components/brave_shields/resources/panel/components/advanced-controls-content/index.tsx:20]: URLs need fixing

File: components/brave_shields/resources/panel/components/main-panel/index.tsx [hunk 28]
```diff
@@ -0,0 +1,51 @@
...
+      </S.PanelHeader>
+      <S.ToggleBox>
+        <FabulouslyLargeToggle
+          brand='shields'
+          accessibleLabel={getLocale('braveShieldsEnable')}
+        />
+        <S.StatusText>
+          {getLocale('braveShields')}
+          {' '}
+          <span>{getLocale('braveShieldsUp').toUpperCase()}</span>
```
Comment (petemill) [inline components/brave_shields/resources/panel/components/main-panel/index.tsx:24]: Should this be a single string so that the syntax order of the UP and DOWN relative to the rest of the sentence makes sense for the specific language?

File: components/brave_shields/resources/panel/components/main-panel/index.tsx [hunk 29]
```diff
@@ -0,0 +1,51 @@
...
+        </S.StatusText>
+        <S.BlockCountBox>
+          <S.BlockCount>21</S.BlockCount>
+          <S.BlockNote>
+            {getLocale('braveShieldsBlockedNote')}
+            {' '}
+            <a href="#">{getLocale('braveShieldsLearnMore')}</a>
```
Comment (petemill) [inline components/brave_shields/resources/panel/components/main-panel/index.tsx:31]: Ditto for the order of the words

File: components/brave_shields/resources/panel/container.tsx [hunk 30]
```diff
@@ -0,0 +1,11 @@
...
+
+import MainPanel from './components/main-panel'
+
+function Container () {
+  return (
+    <MainPanel />
+  )
+}
+
+export default Container
```
Comment (petemill) [inline components/brave_shields/resources/panel/container.tsx:11]: What's the point of this file? Why not just have the consumer render `<MainPanel />` directly?

File: components/web-components/toggle/index.tsx [hunk 31]
```diff
@@ -0,0 +1,58 @@
...
+import * as S from './style'
+
+export type Size = 'sm' | 'default'
+export type Brand = 'vpn' | 'shields'
+interface ToggleProps {
+  size?: Size
+  isOn?: boolean
+  disabled?: boolean
+  accessibleLabel?: string
+  onChange?: (isOn?: boolean) => unknown
```
Comment (petemill) [inline components/web-components/toggle/index.tsx:12]: nit: `isOn` isn't optional, it will always be a `boolean`, right?

**Review by petemill — CHANGES_REQUESTED** [review]
Everything looks good. I only have those questions about the strings.

File: components/brave_shields/resources/panel/container.tsx (hunk 30 above)
Comment (nullhook) [reply components/brave_shields/resources/panel/container.tsx:11]: this file is very similar to the one in vpn. It will be used to render various view types.

File: components/web-components/toggle/index.tsx (hunk 31 above)
Comment (nullhook) [reply components/web-components/toggle/index.tsx:12]: this one is used for default value upon component render

File: components/brave_shields/resources/panel/components/main-panel/index.tsx (hunk 28 above)
Comment (nullhook) [reply components/brave_shields/resources/panel/components/main-panel/index.tsx]: done

File: components/brave_shields/resources/panel/components/main-panel/index.tsx (hunk 29 above)
Comment (nullhook) [reply components/brave_shields/resources/panel/components/main-panel/index.tsx]: makes sense

File: components/web-components/toggle/index.tsx (hunk 31 above)
Comment (petemill) [reply components/web-components/toggle/index.tsx:12]: when `onChange` is called it's always passed a boolean isn't it? Are we talking about the same thing?

File: components/brave_shields/resources/panel/components/main-panel/index.tsx [hunk 32]
```diff
@@ -0,0 +1,51 @@
...
+        <S.SiteTitle>brave.com</S.SiteTitle>
+      </S.PanelHeader>
+      <S.ToggleBox>
+        <FabulouslyLargeToggle
+          brand='shields'
+          accessibleLabel={getLocale('braveShieldsEnable')}
+        />
+        <S.StatusText>
+          {braveShieldsUp.beforeTag}
+          <span>{braveShieldsUp.duringTag}</span>
```
Comment (petemill) [inline components/brave_shields/resources/panel/components/main-panel/index.tsx:25]: missing `.afterTag` usage

File: components/brave_shields/resources/panel/components/main-panel/index.tsx [hunk 33]
```diff
@@ -0,0 +1,51 @@
...
+        </S.StatusText>
+        <S.BlockCountBox>
+          <S.BlockCount>21</S.BlockCount>
+          <S.BlockNote>
+            {braveShieldsBlockedNote.beforeTag}
+            <a href="#">{braveShieldsBlockedNote.duringTag}</a>
```
Comment (petemill) [inline components/brave_shields/resources/panel/components/main-panel/index.tsx:31]: missing `.afterTag` usage

File: components/brave_shields/resources/panel/components/advanced-controls-content/style.ts [hunk 34]
```diff
@@ -0,0 +1,97 @@
...
+  padding: 0;
+  display: inline-flex;
+  align-items: center;
+  font-family: ${(p) => p.theme.fontFamily.heading};
+  font-size: 13px;
+  font-weight: 600;
+  color: ${(p) => p.theme.color.interactive05};
+  width: 48px;
+
+  svg {
```
Comment (petemill) [inline components/brave_shields/resources/panel/components/advanced-controls-content/style.ts:86]:
```suggestion
  ${MyIcon} {
```

**Review by petemill — APPROVED** [review]

File: components/brave_shields/resources/panel/theme/shields-theme.ts [hunk 35]
```diff
@@ -0,0 +1,6 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+export default interface IThemeShields {}
```
Comment (petemill) [inline components/brave_shields/resources/panel/theme/shields-theme.ts:6]: can remove for now
